ci/test-stable.sh

```diff
@@ -58,6 +58,20 @@ test-stable-sbf)
   # SBF solana-sdk legacy compile test
   "$cargo_build_sbf" --manifest-path sdk/Cargo.toml
 
+  # Ensure the minimum supported "rust-version" matches platform tools to fail
+  # quickly if users try to build with an older platform tools install
+  cargo_toml=sdk/program/Cargo.toml
+  source "scripts/read-cargo-variable.sh"
+  crate_rust_version=$(readCargoVariable rust-version $cargo_toml)
+  platform_tools_rust_version=$(sdk/sbf/dependencies/platform-tools/rust/bin/rustc --version)
+  platform_tools_rust_version=$(echo "$platform_tools_rust_version" | cut -d\  -f2) # Remove "rustc " prefix from a string like "rustc 1.68.0-dev"
+  platform_tools_rust_version=$(echo "$platform_tools_rust_version" | cut -d- -f1)  # Remove "-dev" suffix from a string like "1.68.0-dev"
```

Comment:
ftr: I broke this up into multiple lines because bash comments and line continuations have conflicting syntax 😭 

eg:
```bash
foo=$(this \  # THIS COMMENT WON'T WORK
  and # AND ALSO THIS COMMENT WON'T WORK \
  that \
)

Reply:
https://github.com/solana-labs/solana/pull/32254 is looking nice, I'll probably rebase this sucker on that PR once it lands and that'll clean up all this ugly bash!

Reply:
Done, now using the new fancy output from `cargo build-sbf --version`. Thanks @dmakarov!

@brooksprumo or @yihau - this is ready for review now 🥺